### DISCUSSION TITLE
Integration MAVLink on crazyflie

### DISCUSSION THREAD

**Discussion by tchlandry** [bitcraze/discussions#1469] [General]
Hello everyone

I try to implement mavlink protocol on crazyflie 2.1  firmware. In my project I need to send message between crazyflie and raspberry pi zero 2 w. Someone knows how can I do that ? The firmware of drone was modified but the drone doesn't send any mavlink's message. I can't see the logs event, I use tcpdump to capture packet but nothing. Is it possible to do it without PX4 firmware ?

**Comment by evoggy** [comment]
Hi,

The Crazyflie firmware doesn't have any support for MavLink, but there's partial support for PX4 on the Crazyflie hardware. Could you tell us a bit more about what software/firmware you're using?

**Reply by knmcguire** [reply]
Hi! I think you are supposed to flash a different bootloader for the crazyflie px4 firmware as the stock one is not compatible.

Have you found these instructions already? https://docs.px4.io/main/en/complete_vehicles_mc/crazyflie21.html

**Reply by tchlandry** [reply]
Hi thank u for responding
I use this instructions but I have another issue


I recently flashed the PX4 firmware on my Crazyflie 2.1 following the instructions from the [PX4 documentation](https://docs.px4.io/main/en/complete_vehicles_mc/crazyflie21.html). However, after completing the process, I encountered a problem where one of the motors spins continuously as soon as the drone is powered on. 

Here are the steps I followed:

1. **Installed the required tools:**
   
   sudo apt-get update
   sudo apt-get install git python3 python3-pip python3-venv cmake build-essential gcc-arm-none-eabi
   git clone https://github.com/PX4/PX4-Autopilot.git --recursive
   cd PX4-Autopilot
   bash ./Tools/setup/ubuntu.sh
   
   Added the path to `.bashrc`:
   
   export PATH=$PATH:/home/bob/.local/bin
   source ~/.bashrc
   

2. **Cloned PX4 bootloader and created symbolic link to Python 3:**
   
   git clone https://github.com/PX4/Bootloader.git --recurse-submodules
   cd Bootloader
   sudo ln -s /usr/bin/python3 /usr/bin/python
   make crazyflie21_bl
   

3. **Downloaded PX4 Autopilot source and compiled:**
   
   cd ../PX4-Autopilot/
   make bitcraze_crazyflie21_default upload
   

4. **Put the Crazyflie 2.1 in DFU mode and installed `dfu-util`:**
   
   sudo apt-get update
   sudo apt-get install dfu-util
   

5. **Flashed the bootloader:**
   
   sudo dfu-util -d 0483:df11 -a 0 -s 0x08000000 -D ./build/bitcraze_crazyflie21_default/bitcraze_crazyflie21_default.bin
   

6. **Allowed USB port access for Crazyradio:**
   
   sudo groupadd plugdev
   sudo usermod -a -G plugdev $USER
   cat <<EOF | sudo tee /etc/udev/rules.d/99-bitcraze.rules > /dev/null
   # Crazyradio (normal operation)
   SUBSYSTEM=="usb", ATTRS{idVendor}=="1915", ATTRS{idProduct}=="7777", MODE="0664", GROUP="plugdev"
   # Bootloader
   SUBSYSTEM=="usb", ATTRS{idVendor}=="1915", ATTRS{idProduct}=="0101", MODE="0664", GROUP="plugdev"
   # Crazyflie (over USB)
   SUBSYSTEM=="usb", ATTRS{idVendor}=="0483", ATTRS{idProduct}=="5740", MODE="0664", GROUP="plugdev"
   EOF
   sudo udevadm control --reload-rules
   sudo udevadm trigger
  

7. **Compiled and uploaded the firmware to the Crazyflie:**
   
   cd ../PX4-Autopilot/
   make bitcraze_crazyflie21_default upload
   

After following these steps, one of the motors starts spinning continuously once the Crazyflie is powered on. I have verified the hardware connections and retried the steps multiple times but the issue persists.

**Questions:**
- Has anyone experienced similar issues with Crazyflie 2.1 and PX4?
- Are there any specific parameters or configurations that need adjustment to resolve this?
- Could this be a firmware bug or a hardware compatibility issue?

Any help or guidance would be greatly appreciated. Thank you!

**Reply by knmcguire** [reply]
Just for context for others, the same question has been asked here on the PX4 forum: https://discuss.px4.io/t/issue-with-constantly-spinning-motor-on-crazyflie-2-1-after-flashing-px4-firmware/39823

That is perhaps the better community to ask it to. Also drone code foundation has a discord server you can try out as well
